File: wiz-shop-java/src/main/java/com/example/wizshopjava/api/product/dto/ProductTitleResponse.java
package com.example.wizshopjava.api.product.dto;

import java.time.LocalDateTime;

import com.example.wizshopjava.domain.product.entity.Product;
import com.example.wizshopjava.domain.product.entity.ProductStatus;

public record ProductTitleResponse(
	Long productId,
	String productName,
	Long price,
	Long hit,
	ProductStatus status,
	LocalDateTime createdAt
) {
	public static ProductTitleResponse from(Product product) {
		return new ProductTitleResponse(
			product.getId(),
			product.getName(),
			product.getPrice(),
			product.getHit(),
			product.getStatus(),
			product.getCreatedAt()
		);
	}
}

File: wiz-shop/src/main/kotlin/com/example/wizshop/api/product/dto/ProductTitleResponse.kt
package com.example.wizshop.api.product.dto

import com.example.wizshop.domain.product.entity.Product
import com.example.wizshop.domain.product.entity.ProductStatus
import java.time.LocalDateTime

data class ProductTitleResponse(
    val productId: Long,
    val productName: String,
    val price: Long,
    val hit: Long,
    val status: ProductStatus,
    val sellerNickname: String,
    val createdAt: LocalDateTime
) {

    companion object {
        fun from(product: Product) = ProductTitleResponse(
            productId = product.id!!,
            productName = product.name,
            price = product.price,
            hit = product.hit,
            status = product.status,
            sellerNickname = product.seller.nickname,
            createdAt = product.createdAt
        )
    }
}